github/github.go

```diff
@@ -478,12 +478,7 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Res
 		return nil, err
 	}
 
-	defer func() {
-		// Drain up to 512 bytes and close the body to let the Transport reuse the connection
-		io.CopyN(ioutil.Discard, resp.Body, 512)
-		resp.Body.Close()
-	}()
-
+	resp.Body.Close()
```

Comment:
@diorahman This still needs a correction.

As @gmlewis mentioned, this should be same as how this looked prior to the addition of the drain:
https://github.com/google/go-github/commit/e0ff7111b024fda991e457d975e0b73e1380f05f

That means, you should `defer` this statement so that it executes only when the relevant routine returns.

Please change it to:

```go
defer resp.Body.Close()
```